LeaderboardBackend/Services/Impl/LeaderboardService.cs

```diff
@@ -42,4 +43,28 @@ public async Task<CreateLeaderboardResult> CreateLeaderboard(CreateLeaderboardRe
 
         return lb;
     }
+
+    public async Task<RestoreLeaderboardResult> RestoreLeaderboard(long id)
+    {
+        Leaderboard? lb = await applicationContext.Leaderboards.FindAsync([id]);
+
+        if (lb == null)
+        {
+            return new LeaderboardNotFound();
+        }
+
+        if (lb.DeletedAt == null)
+        {
+            return new LeaderboardNeverDeleted();
+        }
+
+        applicationContext.Leaderboards.Update(lb);
+
+        lb.UpdatedAt = clock.GetCurrentInstant();
```

Comment:
If this is broken we should fix it instead of just doing it manually. I can look at it later.